### PR TITLE
Added open id connect configuration for authentication and a custom authorization handler

### PR DESCRIPTION
**Pull request by amitxagarwal**
Added open id connect configuration for authentication and a custom authorization handler based on audience and tenants in the claims.

### CODE REVIEW

File: src/Kmd.Momentum.Mea.Api/Startup.cs [hunk 1]
```diff
@@ -40,22 +44,42 @@ public void ConfigureServices(IServiceCollection services)
...
+
+            var tokenValidationParamteres = new TokenValidationParameters
+            {
+                ValidateIssuer = true,
+                ValidateLifetime = true,
+                ValidateAudience = true,
+                ValidateIssuerSigningKey = true,
+                ValidAudiences = new[]
+                {
+                    "1d18d151-5192-47f1-a611-efa50dbdc431", // application id or client id
```
Comment (ExploreMyCode) [inline src/Kmd.Momentum.Mea.Api/Startup.cs]: As this ApplicationID will change based on TenantID ( 159, 189).  So, we need to come up with way how to dynamically use Application ID as validation parameters. 

Off course, we don't need to solve the problem now, but need to solve it later.

File: src/Kmd.Momentum.Mea.Common/Authorization/MeaCustomClaimHandler.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Kmd.Momentum.Mea.Common.Authorization
+{
+    public class MeaCustomClaimHandler : AuthorizationHandler<MeaCustomClaimRequirement>
+    {
+        public const string Aud = "69d9693e-c4b7-4294-a29f-cddaebfa518b";
+        public readonly List<string> Tenant = new List<string> { "159","189" };
```
Comment (ExploreMyCode) [inline src/Kmd.Momentum.Mea.Common/Authorization/MeaCustomClaimHandler.cs]: Just one thought came in mind and needn't to be solve right now. What will happen - when we have multiple Kommune ID per environment. That time validation will be more complex and need to change the logic.

Comment (amitxagarwal) [reply src/Kmd.Momentum.Mea.Common/Authorization/MeaCustomClaimHandler.cs]: Sure sanjay. .  We need database for that first. .  Once we are done with database design and database has some example data we can further change this. And by that time we will have al the kommune id data and we will share it with tim and adam and then they can configure.

File: src/Kmd.Momentum.Mea.Api/Startup.cs (hunk 1 above)
Comment (amitxagarwal) [reply src/Kmd.Momentum.Mea.Api/Startup.cs]: Yes ..  later in these will come from database right now whether it come from app settings or hard coded it is same for us as if any change is required we have ti rebuild the code again.

File: src/Kmd.Momentum.Mea/MeaHttpClientHelper/IHttpClientHelper.cs [hunk 3]
```diff
@@ -1,11 +1,12 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace Kmd.Momentum.Mea.MeaHttpClientHelper
 {
     public interface IHttpClientHelper
```
Comment (ExploreMyCode) [inline src/Kmd.Momentum.Mea/MeaHttpClientHelper/IHttpClientHelper.cs]: Just Suggestion - If IHttpClientHelper is wired to Citizen then can we put the name something like IHttpClientCitizenHelper. If not then we need to develop common helper let that to handle Citizen or Caseworker ?

What do you think.

Comment (amitxagarwal) [reply src/Kmd.Momentum.Mea/MeaHttpClientHelper/IHttpClientHelper.cs]: ok. I have changed it to citizenhttpclienthelper now

**Pull request merged**